common/src/main/java/org/wso2/testgrid/common/config/JobConfig.java

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package org.wso2.testgrid.common.config;
+
+import org.apache.commons.collections4.ListUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+import static org.wso2.testgrid.common.TestGridConstants.ALL_ALGO;
+import static org.wso2.testgrid.common.TestGridConstants.AT_LEAST_ONE_ALGO;
+import static org.wso2.testgrid.common.TestGridConstants.EXACT_ALGO;
+
+/**
+ * This describes a job configurations.
+ * <p>
+ * The job configuration contain set of job triggering schedules.
+ * Testgrid need to know which schedule should run and which infrastructure
+ * combinations should use for each scheduled job triggers.
+ * This components will configure following schedules in  the TestGrid.
+ * <p>
+ * <li>
+ * <ul>1. Daily</ul>
+ * <ul>2. Weekly</ul>
+ * <ul>3. Monthly</ul>
+ * <ul>4. Manually</ul>
+ * </li>
+ * <p>
+ * Hence, this class represents the JobConfig element of the testgrid.yaml.
+ */
+public class JobConfig implements Serializable {
+
+    private static final Logger logger = LoggerFactory.getLogger(JobConfig.class);
+    private static final long serialVersionUID = -2397625855728044715L;
+
+    private List<Build> builds;
+
+    public JobConfig() {
+
+        this(Collections.emptyList());
+    }
+
+    private JobConfig(List<Build> builds) {
+
+        this.builds = builds;
+    }
+
+    public List<Build> getBuilds() {
+
+        return ListUtils.emptyIfNull(builds);
+    }
+
+    public void setBuilds(
+            List<Build> builds) {
+
+        this.builds = builds;
+    }
+
+    /**
+     * Describes a given build within a job configuration.
+     * A build config should describe its schedule,
+     * and what combinations are there that can be tested.
+     */
+    public static class Build implements Serializable {
+
+        private static final long serialVersionUID = 8474784500779659241L;
+
+        private String schedule;
```

Comment:
What's the default schedle if the user does not specify one?

Reply:
Default schedule is parameterized in TestGridConstants.java file and it is denied as "manual".